supabase/functions/oauth/access-token.ts

```diff
@@ -11,35 +11,42 @@ interface OauthSettings {
 }
 
 export async function accessToken(req: Record<string, any>) {
-    const { state, code_verifier, config, redirect_uri, ...params } = req;
+    const { state, code_verifier, config, redirect_uri, connector_config, ...params } = req;
```

Comment:
I _think_ this is probably OK, but I want to discuss a particular danger so that we can collectively give this a little more scrutiny.

In this context, `connector_config` really only refers to the oauth-related parts. You can either pass your own oauth configs, or use whatever we have in the database for that connector. One of the most important configurations is the connector `endpoint_spec_schema`, which determines which fields are sensitive and need encrypted. For example, if we allowed clients to pass a custom `endpoint_spec_schema` while using "secret" configuration from the database, then the secrets could be leaked. Again, I don't see how that would be possible given the current code. But it might be good to identify and maybe comment the specific invariants that prevent that from happening.

AFAICT, the main thing is really just that we cannot allow any request to provide a custom `endpoint_spec_schema`, but LMK if you've thought of anything else.

Reply:
This sounds right, I can add a comment to `encrypt-config.ts`. FWIW, this PR doesn't actually modify `encrypt-config.ts` at all, so the only `endpoint_spec_schema` we can load must come from the database.

Reply:
FWIW we do also have some basic sanity checking that makes sure we don't expose the plaintext client ID or secret

Reply:
Sounds good, thanks.